### PR TITLE
Update serde dependency to 0.9.0

### PR DESCRIPTION
**Pull request by Marwes**
This lets jsonrpc build on stable (1.15) without needing serde_codegen

### CODE REVIEW

**Comment by tomusdrw** [comment]
Closes #16

File: core/src/lib.rs [hunk 1]
```diff
@@ -22,12 +22,11 @@
 //! ```
 
 #![warn(missing_docs)]
-#![cfg_attr(feature="nightly", feature(custom_derive, plugin))]
-#![cfg_attr(feature="nightly", plugin(serde_macros))]
 
 #[macro_use] extern crate log;
 extern crate serde;
 extern crate serde_json;
+#[macro_use] extern crate serde_derive;
```
Comment (rphmeier) [inline core/src/lib.rs]: style: would prefer to keep `#[macro_use]` links adjacent

**Comment by rphmeier** [comment]
tiny style nit but LGTM. Out of curiosity, how does this affect compile times?

**Comment by Marwes** [comment]
> Out of curiosity, how does this affect compile times?

Running `time cargo test -p jsonrpc-core` gives the same times for me as before this change.

**Comment by rphmeier** [comment]
All good then, thanks!

**Comment by debris** [comment]
side note, we'll been quite substantial refactoring to switch `parity` to `serde 0.9`

**Comment by Marwes** [comment]
When do you think you can release a version with this PR? Using a git dependency right now since I can't use serde 0.8.0 due to another dependency. It works for development but I can't do an actual release to crates.io with it.

**Pull request merged**